cassandra/cluster.py

```diff
@@ -1076,6 +1076,43 @@ def __init__(self,
             self.status_event_refresh_window,
             schema_metadata_enabled, token_metadata_enabled)
 
+    def _create_thread_pool_executor(self, **kwargs):
+        """
+        Create a ThreadPoolExecutor for the cluster. In most cases, the built-in
+        `concurrent.futures.ThreadPoolExecutor` is used.
+
+        Python 3.7 and Eventlet cause the `concurrent.futures.ThreadPoolExecutor`
+        to hang indefinitely. In that case, the user needs to have the `futurist`
+        package so we can use the `futurist.GreenThreadPoolExecutor` class instead.
+
+        :param kwargs: All keyword args are passed to the ThreadPoolExecutor constructor.
+        :return: A ThreadPoolExecutor instance.
+        """
+        tpe_class = ThreadPoolExecutor
+        if sys.version_info[0] == 3 and sys.version_info[1] >= 7:
+            try:
```

Comment:
Possibly use the internal utility function?: https://github.com/datastax/python-driver/blob/master/cassandra/cluster.py#L80-L84

Reply:
The reason why I do it like that is because the user can use eventlet in two ways:
1- monkey patching in their app before creating a cluster. In that case, the utility function would work.
2- Only specifying the connection_class at the cluster level, which will take care to monkey patch when the reactor is initialized. In that case, the utility function wont work since nothing is patched yet. In fact... that might be a oversight cause the documentation mentions that the implementation assume that all eventlet monkey patching is active but the code call monkey_patch. The GeventConnection in example doesn't care of to monkey patch at all.

Reply:
I agree. I will create a ticket to remove that.